File: thain-server/src/main/java/com/xiaomi/thain/server/dao/FlowDao.java
/*
 * Copyright (c) 2019, Xiaomi, Inc.  All rights reserved.
 * This source code is licensed under the Apache License Version 2.0, which
 * can be found in the LICENSE file in the root directory of this source tree.
 */

package com.xiaomi.thain.server.dao;

import com.xiaomi.thain.common.model.FlowModel;
import com.xiaomi.thain.common.model.JobModel;
import com.xiaomi.thain.server.model.sp.FlowListSp;
import com.xiaomi.thain.server.mapper.FlowMapper;
import lombok.NonNull;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Optional;
import java.util.Set;
import javax.annotation.Nullable;

/**
 * @author liangyongrui@xiaomi.com
 */
@Repository
public class FlowDao {

    @NonNull
    private final FlowMapper flowMapper;

    public FlowDao(@NonNull FlowMapper flowMapper) {
        this.flowMapper = flowMapper;
    }

    public boolean getAccessible(long flowId, @NonNull String userId, @Nullable Set<String> appIds) {
        return flowMapper.getUserAccessible(flowId, userId, appIds);
    }

    public List<FlowModel> getFlowList(@NonNull FlowListSp flowListSp) {
        return flowMapper.getFlowList(flowListSp);
    }

    public Long getFlowListCount(@NonNull FlowListSp flowListSp) {
        return flowMapper.getFlowListCount(flowListSp);
    }

    public boolean flowExist(long flowId) {
        return flowMapper.flowExist(flowId);
    }

    public void updateAppId(long flowId, @NonNull String appId) {
        flowMapper.updateAppId(flowId, appId);
    }

    public Optional<FlowModel> getFlow(long flowId) {
        return Optional.ofNullable(flowMapper.getFlow(flowId));
    }

    public List<JobModel> getJobModelList(long flowId) {
        return flowMapper.getJobModelList(flowId);
    }

    public boolean getAccessible(long flowId, @NonNull String appId) {
        return flowMapper.getAppIdAccessible(flowId, appId);
    }
}

File: thain-server/src/main/kotlin/com/xiaomi/thain/server/dao/FlowDao.kt
package com.xiaomi.thain.server.dao

import com.xiaomi.thain.common.model.dr.FlowDr
import com.xiaomi.thain.common.model.dr.JobDr
import com.xiaomi.thain.server.mapper.FlowMapper
import com.xiaomi.thain.server.model.sp.FlowListSp
import org.springframework.stereotype.Repository

/**
 * @author liangyongrui@xiaomi.com
 */
@Repository
class FlowDao(private val flowMapper: FlowMapper) {
    fun getAccessible(flowId: Long, userId: String, appIds: Set<String>?): Boolean {
        return flowMapper.getUserAccessible(flowId, userId, appIds)
    }

    fun getFlowList(flowListSp: FlowListSp): List<FlowDr> {
        return flowMapper.getFlowList(flowListSp)
    }

    fun getFlowListCount(flowListSp: FlowListSp): Long {
        return flowMapper.getFlowListCount(flowListSp)
    }

    fun flowExist(flowId: Long): Boolean {
        return flowMapper.flowExist(flowId)
    }

    fun updateAppId(flowId: Long, appId: String) {
        flowMapper.updateAppId(flowId, appId)
    }

    fun getFlow(flowId: Long): FlowDr? {
        return flowMapper.getFlow(flowId)
    }

    fun getJobModelList(flowId: Long): List<JobDr> {
        return flowMapper.getJobModelList(flowId)
    }

    fun getAccessible(flowId: Long, appId: String): Boolean {
        return flowMapper.getAppIdAccessible(flowId, appId)
    }

}
